src/components/Filters/SchemaSieve.ts

```diff
@@ -44,21 +42,28 @@ export const createFullTextSearchFilter = (
 	schema: JSONSchema,
 	term: string,
 ) => {
-	const stringKeys = reduce(
-		schema.properties,
-		(carry, item: JSONSchema, key) => {
+	const items = Object.entries(schema.properties as JSONSchema).reduce(
+		(itemsAccumulator, entry) => {
+			const [key, item] = entry;
 			if (
 				item.type === 'string' ||
-				(Array.isArray(item.type) && includes(item.type, 'string'))
+				item.type === 'number' ||
+				(Array.isArray(item.type) &&
+					(item.type.includes('string') || item.type.includes('number')))
 			) {
-				carry.push(key);
+				itemsAccumulator.push({
+					key,
+					type: item.type === 'number' ? 'number' : 'string',
+				});
 			}
-
-			return carry;
+			return itemsAccumulator;
 		},
-		[] as string[],
+		[] as Array<{ key: string; type: string }>,
 	);
 
+	const filteredItems = isNaN(Number(term))
+		? items.filter((i) => i.type !== 'number')
```

Comment:
This won't work if `type` is an array of numbers, for example, it's something to try out and write tests/story for.

Reply:
oh yeah, I've to add also the check on the `itemsAccumulator`. Btw sure I'm going to add some test cases, sorry I forgot. @sradevski